Makefile

```diff
@@ -53,6 +53,11 @@ TIMESTAMP:=$${__TIMESTAMP:-$(shell /bin/date -u "+%Y%m%d%H%MZ")}
 SHORT_SHA=`git rev-parse --short HEAD`
 BRANCH=`git rev-parse --abbrev-ref HEAD | tr -- '-/' '_'`
 
+# In case the top commit Depends-On some framework patch that is not merged yet - override framework master with the
+# proper version
+REQ_LEAPP_PR=$(shell git show HEAD -q | grep Depends-On | grep -Eo '[[:digit:]]*')
```

Comment:
The check for `Depends-On` keyword only in top commit of the PR is somewhat cumbersome, what if the PR author:
1. needs to add another commit? He has to add this keyword to every commit message.
2. forgets to add this keyword to the last comment? He has to add another commit or rebase with modified commit message.
3. wnats to change (or even remove) the PR dependency? Same as above

It seems to me that better approach would be querying API and checking PR description for this keyword. This way the PR author (or even repo owner) can easily add/remove the dependecy and also the dependency has better visibility. If implemented this way I would also prefer to have this dependency on its own line. e.g. checking something like `^Depends-On: [[:digit:]]*$`

Reply:
1. That's true, but tbh I see no real problem with this - if 2 or more commits are part of some big change saying clearly that they actually are related to it is common practice. 
2. As we work on feature branches any issue like this is resolved with `git commit -- amend`\`git rebase -i HEAD~2` and a force push. 
3. Same as 2 - just fix the commit message.

But I really liked your idea about github api and even tried to implement it. Unfortunately I had to revert to the old "check commit message for depends-on" behavior because of the following concerns:
- checking gihub api requires the code to be already pushed to the pr branch. Which makes it impossible to run local checks on committed-but-not-yet-pushed code. Running on not-committed code will succeed though.
- you still need to get the dependency information from somewhere. The easiest way I could think of was by adding a label like "requires-XXX", where XXX is number of repository PR. Though it has nicer visibility and any team member can do it, it requires a manual action which can be easily overlooked.
- github api has a certain limit for unauthorized requests, something like 60 per hour. This might not be an issue for travis (for authorized it is 5000 per hour), but if local runs will heavily use api there is a possibility this unauthorized requests limit can be exhausted. Well, I did hit it yesterday night after some really intensive debugging and testing.

I'd say having a combination of 2 (check api for travis + depend on commit message for local run) is a nice follow up, but I'd like to have the initial version simple and working.

Reply:
Thanks for the regex suggestion, fixed https://github.com/oamg/leapp-repository/pull/621/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R57

Reply:
@fernflower  I agree with @vojtechsokol. The need of the "Depends-On" in every commit keep it in the last one seems not good. Why are you checking just the last commit? Why not to check all commits different from master?

Reply:
Because checking last one is way easier than figuring out what commits are in this pr and implementing that separately for travis and local only run.
Again - that's a nice follow up but we have a blocker for proper unittest run right now.

Reply:
@pirat89 implemented as you suggested (checking all commits between HEAD..master for Depends-On) and looks like it works https://travis-ci.org/github/oamg/leapp-repository/jobs/747420365 :)